### DISCUSSION TITLE
Moving newly spawned windows to top automatically

### DISCUSSION THREAD

**Discussion by lgmys-idemia** [leftwm/leftwm#708] [Q&A]
Hey, I was wondering if it is possible to have new windows moved to top automatically in master + vert stack layout? Kind of what dwm does by default.

Thanks for the great software!

**Comment by lgmys-idemia** [comment]
Looks like there is a way! Add the following to the `theme.toml` file:

`on_new_window = "leftwm-command MoveWindowTop"`

**Reply by VuiMuich** [reply]
Welcome adventurer!

Nice you found this themeing option.
Just for reference, setting this in `config.toml` should have a similar yet slightly different effect:
```toml
focus_new_windows = true
```

**Comment by kid** [comment]
I didn't know about that `on_new_window` 🤔 

there is also #358 which I started working on over the weekend :)

**Reply by lgestc** [reply]
Actually, after a while toying with it... Native option would be much better than executing another command after the window is spawned. It does not feel snappy on my machine. @kid is is far from done? need some help?

**Reply by kid** [reply]
Hoping to finish this over the weekend, it's actually fairly simple to do, as always it's writing tests for it that takes more time :)

**Reply by lgestc** [reply]
Cool! Do not shy away from sharing it once done:).

**Reply by VuiMuich** [reply]
> Hoping to finish this over the weekend, it's actually fairly simple to do, as always it's writing tests for it that takes more time :)

If it you got the mechanics working you could open a draft PR and mark it `Ready for Review` when you added the tests.

**Comment by lgestc** [comment]
In the meantime, I did allow myself to create a PR of my own that adds this functionality :) #709

**Comment by kid** [comment]
Opened my draft PR in #710 :)

**Reply by lgestc** [reply]
What is missing to have it merged with the upstream?:) Can I help?

**Reply by VuiMuich** [reply]
Just a lack of time to review and runtime test. 
If you have a build running on this branch and don't experiencer any bugs or regressions I think we could merge it.

I'll try to get to this later this evening.

**Reply by lgestc** [reply]
It works fine for me, I did post a comment on that in the PR itself today:).
